packages/nimble/R/cppDefs_core.R

```diff
@@ -230,6 +230,10 @@ cppCodeBlock <- setRefClass('cppCodeBlock',
                                           cppADCode = 'ANY', generatorSymTab = 'ANY'),			#'logical'),
                             methods = list(
                                 generate = function(indent = '', ...) {
+                                    if(identical(cppADCode, TRUE)){
+                                      nimbleUserNamespace$cppADCode <- TRUE
+                                      on.exit(nimbleUserNamespace$cppADCode <- FALSE)
```

Comment:
Will this work when blocks are nested? It seems to me that you'd want to avoid setting if it's already true, otherwise you might erroneously set it to FALSE on exit.
```r
if (!nimbleUserNamespace$cppADCode) {
    nimbleUserNamespace$cppADCode <- TRUE
    on.exit(nimbleUserNamespace$cppADCode <- FALSE
}
```

Reply:
Good logical catch @fritzo!  That makes total sense to me.